// _vant@0.11.10@vant/lib/sku/components/SkuActions.js
'use strict';

exports.__esModule = true;

var _button = require('../../button');

var _button2 = _interopRequireDefault(_button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  render: function render() {
    var _vm = this;var _h = _vm.$createElement;var _c = _vm._self._c || _h;return _c('div', { staticClass: "van-sku-actions" }, [_vm.showAddCartBtn ? _c('van-button', { attrs: { "bottomAction": "" }, on: { "click": _vm.onAddCartClicked } }, [_vm._v("加入购物车")]) : _vm._e(), _c('van-button', { attrs: { "type": "primary", "bottomAction": "" }, on: { "click": _vm.onBuyClicked } }, [_vm._v(_vm._s(_vm.buyText))])], 1);
  },

  name: 'van-sku-actions',

  components: {
    VanButton: _button2.default
  },

  props: {
    skuEventBus: Object,
    showAddCartBtn: Boolean,
    buyText: {
      type: String,
      default: '立即购买'
    }
  },

  methods: {
    onAddCartClicked: function onAddCartClicked() {
      this.skuEventBus.$emit('sku:addCart');
    },
    onBuyClicked: function onBuyClicked() {
      this.skuEventBus.$emit('sku:buy');
    }
  }
};
